core/libdeno/internal.h

```diff
@@ -25,6 +26,60 @@ struct ModuleInfo {
   }
 };
 
+class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
+ public:
+  ~ArrayBufferAllocator() {
+    std::lock_guard<std::mutex> lock(ref_count_map_mutex_);
```

Comment:
If there was just a single global Allocator, would the mutex still be needed?

Reply:
Of course, even more so than now/.